Shared/Samples/Show geodesic sector and ellipse/ShowGeodesicSectorAndEllipseView.swift

```diff
@@ -0,0 +1,385 @@
+                                edges: [.top]
+                            )
+                        }
+                        .presentationDetents([.medium])
+                    }
+                }
+            }
+        }
+    }
+}
+
+private extension ShowGeodesicSectorAndEllipseView {
+    /// Custom data type so that Geometry options can be displayed in the menu.
+    enum GeometryType: CaseIterable {
+        case point, polyline, polygon
+        
+        var label: String {
+            switch self {
+            case .point: "Point"
+            case .polyline: "Polyline"
+            case .polygon: "Polygon"
+            }
+        }
+    }
+    
+    /// A view model that encapsulates logic and state for rendering a geodesic sector and ellipse.
+    /// Handles user-configured parameters and updates overlays when those parameters change.
+    @MainActor
+    class Model: ObservableObject {
+        /// The map that will be displayed in the map view.
+        let map = Map(basemapStyle: .arcGISTopographic)
+        
+        /// The map point selected by the user when tapping on the map.
+        @Published var center: Point? {
+            didSet {
+                guard let center else { return }
+                updateSector(center: center)
+            }
+        }
+        
+        var graphicOverlays: [GraphicsOverlay] {
+            return [ellipseGraphicOverlay, sectorGraphicOverlay]
+        }
+        
+        /// The graphics overlay that will be displayed on the map view.
+        /// This will hold the graphics that show the ellipse path.
+        private let ellipseGraphicOverlay = GraphicsOverlay()
+        
+        /// The graphics overlay that will be displayed on the map view.
+        /// This will display a highlighted section of the ellipse path.
+        private let sectorGraphicOverlay = GraphicsOverlay()
+        
+        /// The direction (in degrees) of the ellipse's major axis.
+        @Published var axisDirection: Double = 45
+        /// Controls the complexity of the geometries and the approximation of the ellipse curve.
+        @Published var maxSegmentLength: Double = 1
+        /// Changes the sectors shape.
+        @Published var sectorAngle: Double = 90
+        /// Controls the complexity of the geometries and the approximation of the ellipse curve.
+        @Published var maxPointCount: Int = 1_000
+        /// Changes the length of ellipse shape on one axis.
+        @Published var semiAxis1Length: Double = 200
+        /// Changes the length of ellipse shape on one axis.
+        @Published var semiAxis2Length: Double = 100
+        /// Changes the geometry type which the sector is rendered.
+        @Published var geometryType: GeometryType = .polygon
+        /// Changes the direction of the sector.
+        @Published var startDirection: Double = 45
+        
+        func refreshSector() {
+            guard let center = center else { return }
```

Comment:
I think this can be simplified to `guard let center else { return }`

Reply:
Yes it can!